### DISCUSSION TITLE
Are the foreign keys required?

### DISCUSSION THREAD

**Discussion by Roiskia** [quartznet/quartznet#2289] [General] [answered]
I am working on integrating the Quartz Scheduler into an existing project. Part of this project is a legacy application that is used as a DB manager to install and update the database schema using scripts written with a DSL. We use this application to support multiple DBMS vendors and I am required to use it.

One disadvantage of this DB manager is that it does not support foreign keys. As the quarts schema uses foreign keys, I wanted to ask if they are necessary and if we need to upgrade our DB manager.

**Comment by lahma** [comment, marked as answer]
Well, foreign keys are like seatbelts in a car, they can save you from bad things happening. There's even law enforcement enforcing them in some countries. In this case you your government doesn't and is probably also trying to sabotage your brakes.

Quartz won't check their existence. Make sure you have required indexes in place, unless having them is forbidden too (analog would be you are only allowed to use the car's first gear).

**Reply by Roiskia** [reply]
Thank you for your prompt reply. I am aware of the drawbacks of omitting foreign keys, which is why I asked in the first place. 

Justifying foreign keys as a requirement of a dependency may have been what was needed to get them, as the conventional reasons did not seem sufficient.

Indexes are fine though, so I guess I have to rely on them.
